### ISSUE TITLE
feat: Per-channel ImageNet normalization in Transforms

### ISSUE THREAD

**Issue by Samyssmile** [Samyssmile/edux#197]
## Summary

`Transforms.normalize()` currently only supports uniform normalization (`/255` or single mean/std). ImageNet training requires **per-channel** normalization with the well-known ImageNet statistics. Without it, convergence is significantly slower (~2–3% Top-1 loss).

## Current State

```java
// Transforms.java — only two options:
public static Transform normalize() {
    return input -> input.mul(1.0f / 255.0f);   // [0,255] → [0,1], no channel awareness
}

public static Transform normalize(float mean, float std) {
    return input -> {
        var result = new float[data.length];
        for (int i = 0; i < data.length; i++) {
            result[i] = (data[i] - mean) / std;  // same mean/std for ALL channels
        }
        return FloatTensor.of(input.shape(), result);
    };
}
```

**Problem**: No way to apply different mean/std per RGB channel.

## Why Per-Channel Normalization Matters

ImageNet images have different statistics per channel:
```
R channel: mean=0.485, std=0.229
G channel: mean=0.456, std=0.224
B channel: mean=0.406, std=0.225
```

These values are computed over the entire ImageNet training set. All SOTA models (ResNet, EfficientNet, ViT) use this normalization. Without it:
- Feature distributions are shifted/scaled inconsistently across channels
- BatchNorm must compensate for channel imbalance in early layers
- Convergence is slower, final accuracy is lower

## Proposed API

```java
public final class Transforms {

    // Existing (unchanged)
    public static Transform normalize() { ... }
    public static Transform normalize(float mean, float std) { ... }

    // NEW: Per-channel normalization
    /**
     * Normalizes each channel independently: output[c] = (input[c] - mean[c]) / std[c]
     * Expects input tensor shape [C, H, W] with values in [0, 1].
     *
     * @param mean per-channel means, length must equal number of channels
     * @param std  per-channel standard deviations, length must equal number of channels
     */
    public static Transform normalize(float[] mean, float[] std) {
        return input -> {
            int c = input.shape().get(0);
            int spatial = input.shape().size() / c;  // H * W
            float[] src = input.data();
            float[] result = new float[src.length];

            for (int ch = 0; ch < c; ch++) {
                int offset = ch * spatial;
                float m = mean[ch];
                float s = std[ch];
                for (int i = 0; i < spatial; i++) {
                    result[offset + i] = (src[offset + i] - m) / s;
                }
            }
            return FloatTensor.of(input.shape(), result);
        };
    }

    // NEW: Convenience preset for ImageNet
    /** Standard ImageNet normalization (RGB, values expected in [0, 1]). */
    public static Transform imagenetNormalize() {
        return normalize(
            new float[]{0.485f, 0.456f, 0.406f},
            new float[]{0.229f, 0.224f, 0.225f}
        );
    }
}
```

### Usage
```java
// Training pipeline
var trainTransform = Transforms.chain(
    Transforms.normalize(),              // [0,255] → [0,1]
    Transforms.imagenetNormalize()       // per-channel standardization
);

var dataset = ImageFolderDataset.fromDirectory(
    Path.of("datasets/imagenet"),
    3, 224, 224,
    trainTransform
);
```

## Implementation Notes

- Inner loop over spatial dims is SIMD-friendly (contiguous memory per channel in CHW layout)
- Consider adding a `Transforms.chain(Transform... transforms)` utility for composing transforms
- The `normalize()` (uniform) and `normalize(float[], float[])` (per-channel) should be clearly documented to avoid confusion
- Values are expected in `[0, 1]` after initial `/255` normalization

## Acceptance Criteria

- [ ] `Transforms.normalize(float[] mean, float[] std)` correctly normalizes per channel
- [ ] `Transforms.imagenetNormalize()` provides standard ImageNet statistics
- [ ] Works with [C, H, W] tensors (3-channel RGB)
- [ ] Validates that `mean.length == std.length == numChannels`
- [ ] Optional: SIMD loop for performance
- [ ] Unit test: per-channel output values match expected computation
- [ ] Unit test: ImageNet preset produces correct statistics

**Comment by Samyssmile** [comment]
Fixed: numWorkers now correctly limits parallel sample-loading concurrency via Semaphore, producer thread join on close(), and documented single-consumer contract.
